[FEATURE] Better Empty CSV Messaging

### Is your feature request related to a problem? Please describe

Schemathesis uses `?format=csv` but then tries to decode the empty response `''` as a JSON object.

### Describe the solution you'd like

Better messaging. Say that it's attempting `?format=csv` but then trying to decode as JSON.

### Describe alternatives you've considered

In Django Rest Framework, set `renderer_classes = (JSONRenderer,)` in the ViewSet so that DRF-Spectacular is not advertising CSV support and DRF is not supporting it at all.

DRF by default supports a JSON Renderer and a CSV Renderer.

### Additional context

Include any other context, like use-cases, diagrams, or screenshots, to help us understand the feature request.

```python
auth_headers = {'Authorization': 'Token <redacted>', 'User-Agent': 'schemathesis/3.39.8'}

    @wraps(test)
>   def test_function(*args: Any, **kwargs: Any) -> Any:
E   schemathesis.exceptions.CheckFailed: Schemathesis found 2 distinct sets of failures.
E   ====================
E   
E   self = <OpenApi30 for Ad API v1 (v1)>
E   operation = APIOperation(path='/api/endpoint', method='get', schema=<OpenApi30 for API v1 (v1)>, verbose_name='GET /a....', 'schema': {'type': 'integer'}})]), body=PayloadAlternatives(items=[]), case_cls=<class 'schemathesis.models.Case'>)
E   response = <Response [200]>
E   
E       def validate_response(self, operation: APIOperation, response: GenericResponse) -> bool | None:
E           responses = {str(key): value for key, value in operation.definition.raw.get("responses", {}).items()}
E           status_code = str(response.status_code)
E           if status_code in responses:
E               definition = responses[status_code]
E           elif "default" in responses:
E               definition = responses["default"]
E           else:
E               # No response defined for the received response status code
E               return None
E           scopes, schema = self.get_response_schema(definition, operation.definition.scope)
E           if not schema:
E               # No schema to check against
E               return None
E           content_type = response.headers.get("Content-Type")
E           errors = []
E           if content_type is None:
E               media_types = self.get_content_types(operation, response)
E               formatted_content_types = [f"\n- `{content_type}`" for content_type in media_types]
E               message = f"The following media types are documented in the schema:{''.join(formatted_content_types)}"
E               try:
E                   raise get_missing_content_type_error(operation.verbose_name)(
E                       failures.MissingContentType.title,
E                       context=failures.MissingContentType(message=message, media_types=media_types),
E                   )
E               except Exception as exc:
E                   errors.append(exc)
E           if content_type and not is_json_media_type(content_type):
E               _maybe_raise_one_or_more(errors)
E               return None
E           try:
E   >           data = get_json(response)
E   
E   ../../../lib/python3.11/site-packages/schemathesis/specs/openapi/schemas.py:680: 
E   _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
E   ../../../lib/python3.11/site-packages/schemathesis/transports/responses.py:43: in get_json
E       return json.loads(response.text)
E   ../../../lib/python3.11/json/__init__.py:346: in loads
E       return _default_decoder.decode(s)
E   ../../../lib/python3.11/json/decoder.py:337: in decode
E       obj, end = self.raw_decode(s, idx=_w(s, 0).end())
E   _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
E   
E   self = <json.decoder.JSONDecoder object at 0x7dab076c0e50>, s = '', idx = 0
E   
E       def raw_decode(self, s, idx=0):
E           """Decode a JSON document from ``s`` (a ``str`` beginning with
E           a JSON document) and return a 2-tuple of the Python
E           representation and the index in ``s`` where the document ended.
E       
E           This can be used to decode a JSON document from a string that may
E           have extraneous data at the end.
E       
E           """
E           try:
E               obj, end = self.scan_once(s, idx)
E           except StopIteration as err:
E   >           raise JSONDecodeError("Expecting value", s, err.value) from None
E   E           json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
E   
E   ../../../lib/python3.11/json/decoder.py:355: JSONDecodeError
E   
```

Do you run Schemathesis via pytest? in CLI the output would have a cURL command which in this case would have the `format=csv` query param:

```
=================================== FAILURES ===================================
_____________________________ GET /malformed_json ______________________________
1. Test Case ID: <PLACEHOLDER>

- JSON deserialization error

    Response must be valid JSON with 'Content-Type: application/json' header:

      Expecting property name enclosed in double quotes: line 1 column 2 (char 1)

[200] OK:

    `{malformed}`

Reproduce with:

    curl -X GET http://127.0.0.1/api/malformed_json
```

(I added an extra line about expectations)

Hi, sorry this took so long to get.

```
auth_headers = {'Authorization': 'Token 0650532ceb4ef1375e45f82ce874897f50884ae9', 'User-Agent': 'schemathesis/3.39.10'}

    @wraps(test)
>   def test_function(*args: Any, **kwargs: Any) -> Any:
E   schemathesis.exceptions.CheckFailed: Schemathesis found 2 distinct sets of failures.
E   ====================
E   
E   self = <OpenApi30 for API v1 (v1)>
E   operation = APIOperation(path=‘/path/‘, method='get', schema=<OpenApi30 for API v1 (v1)>, verbose_name='GET...'})]), body=PayloadAlternatives(items=[]), case_cls=<class 'schemathesis.models.Case'>)
E   response = <Response [200]>
E   
E       def validate_response(self, operation: APIOperation, response: GenericResponse) -> bool | None:
E           responses = {str(key): value for key, value in operation.definition.raw.get("responses", {}).items()}
E           status_code = str(response.status_code)
E           if status_code in responses:
E               definition = responses[status_code]
E           elif "default" in responses:
E               definition = responses["default"]
E           else:
E               # No response defined for the received response status code
E               return None
E           scopes, schema = self.get_response_schema(definition, operation.definition.scope)
E           if not schema:
E               # No schema to check against
E               return None
E           content_type = response.headers.get("Content-Type")
E           errors = []
E           if content_type is None:
E               media_types = self.get_content_types(operation, response)
E               formatted_content_types = [f"\n- `{content_type}`" for content_type in media_types]
E               message = f"The following media types are documented in the schema:{''.join(formatted_content_types)}"
E               try:
E                   raise get_missing_content_type_error(operation.verbose_name)(
E                       failures.MissingContentType.title,
E                       context=failures.MissingContentType(message=message, media_types=media_types),
E                   )
E               except Exception as exc:
E                   errors.append(exc)
E           if content_type and not is_json_media_type(content_type):
E               _maybe_raise_one_or_more(errors)
E               return None
E           try:
E   >           data = get_json(response)
E   
E   ../../../lib/python3.11/site-packages/schemathesis/specs/openapi/schemas.py:680: 
E   _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
E   ../../../lib/python3.11/site-packages/schemathesis/transports/responses.py:43: in get_json
E       return json.loads(response.text)
E   ../../../lib/python3.11/json/__init__.py:346: in loads
E       return _default_decoder.decode(s)
E   ../../../lib/python3.11/json/decoder.py:337: in decode
E       obj, end = self.raw_decode(s, idx=_w(s, 0).end())
E   _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
E   
E   self = <json.decoder.JSONDecoder object at 0x79db53a0cf90>, s = '', idx = 0
E   
E       def raw_decode(self, s, idx=0):
E           """Decode a JSON document from ``s`` (a ``str`` beginning with
E           a JSON document) and return a 2-tuple of the Python
E           representation and the index in ``s`` where the document ended.
E       
E           This can be used to decode a JSON document from a string that may
E           have extraneous data at the end.
E       
E           """
E           try:
E               obj, end = self.scan_once(s, idx)
E           except StopIteration as err:
E   >           raise JSONDecodeError("Expecting value", s, err.value) from None
E   E           json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
E   
E   ../../../lib/python3.11/json/decoder.py:355: JSONDecodeError
E   
E   The above exception was the direct cause of the following exception:
E   
E   self = <OpenApi30 for API v1 (v1)>
E   operation = APIOperation(path='/path/‘, method='get', schema=<OpenApi30 for API v1 (v1)>, verbose_name='GET...'})]), body=PayloadAlternatives(items=[]), case_cls=<class 'schemathesis.models.Case'>)
E   response = <Response [200]>
E   
E       def validate_response(self, operation: APIOperation, response: GenericResponse) -> bool | None:
E           responses = {str(key): value for key, value in operation.definition.raw.get("responses", {}).items()}
E           status_code = str(response.status_code)
E           if status_code in responses:
E               definition = responses[status_code]
E           elif "default" in responses:
E               definition = responses["default"]
E           else:
E               # No response defined for the received response status code
E               return None
E           scopes, schema = self.get_response_schema(definition, operation.definition.scope)
E           if not schema:
E               # No schema to check against
E               return None
E           content_type = response.headers.get("Content-Type")
E           errors = []
E           if content_type is None:
E               media_types = self.get_content_types(operation, response)
E               formatted_content_types = [f"\n- `{content_type}`" for content_type in media_types]
E               message = f"The following media types are documented in the schema:{''.join(formatted_content_types)}"
E               try:
E                   raise get_missing_content_type_error(operation.verbose_name)(
E                       failures.MissingContentType.title,
E                       context=failures.MissingContentType(message=message, media_types=media_types),
E                   )
E               except Exception as exc:
E                   errors.append(exc)
E           if content_type and not is_json_media_type(content_type):
E               _maybe_raise_one_or_more(errors)
E               return None
E           try:
E               data = get_json(response)
E           except JSONDecodeError as exc:
E               exc_class = get_response_parsing_error(operation.verbose_name, exc)
E               context = failures.JSONDecodeErrorContext.from_exception(exc)
E               try:
E   >               raise exc_class(context.title, context=context) from exc
E   E               schemathesis.exceptions.CheckFailed: JSON deserialization error
E   
E   ../../../lib/python3.11/site-packages/schemathesis/specs/openapi/schemas.py:685: CheckFailed
E   
E   During handling of the above exception, another exception occurred:
E   Traceback (most recent call last):
E     File "/usr/local/lib/python3.11/site-packages/schemathesis/specs/openapi/schemas.py", line 685, in validate_response
E       raise exc_class(context.title, context=context) from exc
E   schemathesis.exceptions.CheckFailed: JSON deserialization error
E   
E   During handling of the above exception, another exception occurred:
E   
E     + Exception Group Traceback (most recent call last):
E     |   File "/usr/local/lib/python3.11/site-packages/schemathesis/_hypothesis.py", line 83, in test_function
E     |     return test(*args, **kwargs)
E     |            ^^^^^^^^^^^^^^^^^^^^^
E     |   File "/usr/local/src/webapp/src/api/tests/test_schemathesis.py", line 21, in test_api
E     |     case.call_and_validate(
E     |   File "/usr/local/lib/python3.11/site-packages/schemathesis/models.py", line 591, in call_and_validate
E     |     self.validate_response(
E     |   File "/usr/local/lib/python3.11/site-packages/schemathesis/models.py", line 519, in validate_response
E     |     check(ctx, response, copied_case)
E     |   File "/usr/local/lib/python3.11/site-packages/schemathesis/specs/openapi/checks.py", line 219, in response_schema_conformance
E     |     return case.operation.validate_response(response)
E     |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E     |   File "/usr/local/lib/python3.11/site-packages/schemathesis/models.py", line 903, in validate_response
E     |     return self.schema.validate_response(self, response)
E     |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E     |   File "/usr/local/lib/python3.11/site-packages/schemathesis/specs/openapi/schemas.py", line 688, in validate_response
E     |     _maybe_raise_one_or_more(errors)
E     |   File "/usr/local/lib/python3.11/site-packages/schemathesis/specs/openapi/schemas.py", line 812, in _maybe_raise_one_or_more
E     |     raise MultipleFailures("\n\n".join(str(error) for error in errors), errors)
E     | ExceptionGroup: Missing Content-Type header
E     | 
E     | JSON deserialization error (2 sub-exceptions)
E     +-+---------------- 1 ----------------
E       | Traceback (most recent call last):
E       |   File "/usr/local/lib/python3.11/site-packages/schemathesis/specs/openapi/schemas.py", line 670, in validate_response
E       |     raise get_missing_content_type_error(operation.verbose_name)(
E       | schemathesis.exceptions.CheckFailed: Missing Content-Type header
E       +---------------- 2 ----------------
E       | Traceback (most recent call last):
E       |   File "/usr/local/lib/python3.11/site-packages/schemathesis/specs/openapi/schemas.py", line 680, in validate_response
E       |     data = get_json(response)
E       |            ^^^^^^^^^^^^^^^^^^
E       |   File "/usr/local/lib/python3.11/site-packages/schemathesis/transports/responses.py", line 43, in get_json
E       |     return json.loads(response.text)
E       |            ^^^^^^^^^^^^^^^^^^^^^^^^^
E       |   File "/usr/local/lib/python3.11/json/__init__.py", line 346, in loads
E       |     return _default_decoder.decode(s)
E       |            ^^^^^^^^^^^^^^^^^^^^^^^^^^
E       |   File "/usr/local/lib/python3.11/json/decoder.py", line 337, in decode
E       |     obj, end = self.raw_decode(s, idx=_w(s, 0).end())
E       |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E       |   File "/usr/local/lib/python3.11/json/decoder.py", line 355, in raw_decode
E       |     raise JSONDecodeError("Expecting value", s, err.value) from None
E       | json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
E       | 
E       | The above exception was the direct cause of the following exception:
E       | 
E       | Traceback (most recent call last):
E       |   File "/usr/local/lib/python3.11/site-packages/schemathesis/specs/openapi/schemas.py", line 685, in validate_response
E       |     raise exc_class(context.title, context=context) from exc
E       | schemathesis.exceptions.CheckFailed: JSON deserialization error
E       +------------------------------------
E   
E   ====================
E   
E   self = <OpenApi30 for API v1 (v1)>
E   operation = APIOperation(path='/path’, method='get', schema=<OpenApi30 for API v1 (v1)>, verbose_name='GET...'})]), body=PayloadAlternatives(items=[]), case_cls=<class 'schemathesis.models.Case'>)
E   response = <Response [200]>
E   
E       def validate_response(self, operation: APIOperation, response: GenericResponse) -> bool | None:
E           responses = {str(key): value for key, value in operation.definition.raw.get("responses", {}).items()}
E           status_code = str(response.status_code)
E           if status_code in responses:
E               definition = responses[status_code]
E           elif "default" in responses:
E               definition = responses["default"]
E           else:
E               # No response defined for the received response status code
E               return None
E           scopes, schema = self.get_response_schema(definition, operation.definition.scope)
E           if not schema:
E               # No schema to check against
E               return None
E           content_type = response.headers.get("Content-Type")
E           errors = []
E           if content_type is None:
E               media_types = self.get_content_types(operation, response)
E               formatted_content_types = [f"\n- `{content_type}`" for content_type in media_types]
E               message = f"The following media types are documented in the schema:{''.join(formatted_content_types)}"
E               try:
E                   raise get_missing_content_type_error(operation.verbose_name)(
E                       failures.MissingContentType.title,
E                       context=failures.MissingContentType(message=message, media_types=media_types),
E                   )
E               except Exception as exc:
E                   errors.append(exc)
E           if content_type and not is_json_media_type(content_type):
E               _maybe_raise_one_or_more(errors)
E               return None
E           try:
E   >           data = get_json(response)
E   
E   ../../../lib/python3.11/site-packages/schemathesis/specs/openapi/schemas.py:680: 
E   _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
E   ../../../lib/python3.11/site-packages/schemathesis/transports/responses.py:43: in get_json
E       return json.loads(response.text)
E   ../../../lib/python3.11/json/__init__.py:346: in loads
E       return _default_decoder.decode(s)
E   ../../../lib/python3.11/json/decoder.py:337: in decode
E       obj, end = self.raw_decode(s, idx=_w(s, 0).end())
E   _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
E   
E   self = <json.decoder.JSONDecoder object at 0x79db53a0cf90>, s = '', idx = 0
E   
E       def raw_decode(self, s, idx=0):
E           """Decode a JSON document from ``s`` (a ``str`` beginning with
E           a JSON document) and return a 2-tuple of the Python
E           representation and the index in ``s`` where the document ended.
E       
E           This can be used to decode a JSON document from a string that may
E           have extraneous data at the end.
E       
E           """
E           try:
E               obj, end = self.scan_once(s, idx)
E           except StopIteration as err:
E   >           raise JSONDecodeError("Expecting value", s, err.value) from None
E   E           json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
E   
E   ../../../lib/python3.11/json/decoder.py:355: JSONDecodeError
E   
E   The above exception was the direct cause of the following exception:
E   
E   self = <OpenApi30 for API v1 (v1)>
E   operation = APIOperation(path=‘/path/‘, method='get', schema=<OpenApi30 for API v1 (v1)>, verbose_name='GET...'})]), body=PayloadAlternatives(items=[]), case_cls=<class 'schemathesis.models.Case'>)
E   response = <Response [200]>
E   
E       def validate_response(self, operation: APIOperation, response: GenericResponse) -> bool | None:
E           responses = {str(key): value for key, value in operation.definition.raw.get("responses", {}).items()}
E           status_code = str(response.status_code)
E           if status_code in responses:
E               definition = responses[status_code]
E           elif "default" in responses:
E               definition = responses["default"]
E           else:
E               # No response defined for the received response status code
E               return None
E           scopes, schema = self.get_response_schema(definition, operation.definition.scope)
E           if not schema:
E               # No schema to check against
E               return None
E           content_type = response.headers.get("Content-Type")
E           errors = []
E           if content_type is None:
E               media_types = self.get_content_types(operation, response)
E               formatted_content_types = [f"\n- `{content_type}`" for content_type in media_types]
E               message = f"The following media types are documented in the schema:{''.join(formatted_content_types)}"
E               try:
E                   raise get_missing_content_type_error(operation.verbose_name)(
E                       failures.MissingContentType.title,
E                       context=failures.MissingContentType(message=message, media_types=media_types),
E                   )
E               except Exception as exc:
E                   errors.append(exc)
E           if content_type and not is_json_media_type(content_type):
E               _maybe_raise_one_or_more(errors)
E               return None
E           try:
E               data = get_json(response)
E           except JSONDecodeError as exc:
E               exc_class = get_response_parsing_error(operation.verbose_name, exc)
E               context = failures.JSONDecodeErrorContext.from_exception(exc)
E               try:
E   >               raise exc_class(context.title, context=context) from exc
E   E               schemathesis.exceptions.CheckFailed: JSON deserialization error
E   
E   ../../../lib/python3.11/site-packages/schemathesis/specs/openapi/schemas.py:685: CheckFailed
E   
E   During handling of the above exception, another exception occurred:
E   Traceback (most recent call last):
E     File "/usr/local/lib/python3.11/site-packages/schemathesis/specs/openapi/schemas.py", line 685, in validate_response
E       raise exc_class(context.title, context=context) from exc
E   schemathesis.exceptions.CheckFailed: JSON deserialization error
E   
E   During handling of the above exception, another exception occurred:
E   
E     + Exception Group Traceback (most recent call last):
E     |   File "/usr/local/lib/python3.11/site-packages/schemathesis/lazy.py", line 370, in run_subtest
E     |     sub_test(**fixtures)
E     |   File "/usr/local/lib/python3.11/site-packages/schemathesis/_hypothesis.py", line 81, in test_api
E     |     def test_function(*args: Any, **kwargs: Any) -> Any:
E     |                ^^^^^^^^^^^^^^^
E     |   File "/usr/local/lib/python3.11/site-packages/hypothesis/core.py", line 1834, in wrapped_test
E     |     raise the_error_hypothesis_found
E     | ExceptionGroup: Missing Content-Type header
E     | 
E     | JSON deserialization error (2 sub-exceptions)
E     | Falsifying example: test_api(
E     |     auth_headers={'Authorization': 'Token 0650532ceb4ef1375e45f82ce874897f50884ae9',
E     |      'User-Agent': 'schemathesis/3.39.10'},
E     |     case=,
E     | )
E     | 
E     | You can reproduce this example by temporarily adding @reproduce_failure('6.125.3', b'AIABQQRBAAA=') as a decorator on your test case
E     +-+---------------- 1 ----------------
E       | Traceback (most recent call last):
E       |   File "/usr/local/lib/python3.11/site-packages/schemathesis/specs/openapi/schemas.py", line 670, in validate_response
E       |     raise get_missing_content_type_error(operation.verbose_name)(
E       | schemathesis.exceptions.CheckFailed: Missing Content-Type header
E       +---------------- 2 ----------------
E       | Traceback (most recent call last):
E       |   File "/usr/local/lib/python3.11/site-packages/schemathesis/specs/openapi/schemas.py", line 680, in validate_response
E       |     data = get_json(response)
E       |            ^^^^^^^^^^^^^^^^^^
E       |   File "/usr/local/lib/python3.11/site-packages/schemathesis/transports/responses.py", line 43, in get_json
E       |     return json.loads(response.text)
E       |            ^^^^^^^^^^^^^^^^^^^^^^^^^
E       |   File "/usr/local/lib/python3.11/json/__init__.py", line 346, in loads
E       |     return _default_decoder.decode(s)
E       |            ^^^^^^^^^^^^^^^^^^^^^^^^^^
E       |   File "/usr/local/lib/python3.11/json/decoder.py", line 337, in decode
E       |     obj, end = self.raw_decode(s, idx=_w(s, 0).end())
E       |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E       |   File "/usr/local/lib/python3.11/json/decoder.py", line 355, in raw_decode
E       |     raise JSONDecodeError("Expecting value", s, err.value) from None
E       | json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
E       | 
E       | The above exception was the direct cause of the following exception:
E       | 
E       | Traceback (most recent call last):
E       |   File "/usr/local/lib/python3.11/site-packages/schemathesis/specs/openapi/schemas.py", line 685, in validate_response
E       |     raise exc_class(context.title, context=context) from exc
E       | schemathesis.exceptions.CheckFailed: JSON deserialization error
E       +------------------------------------

../../../lib/python3.11/site-packages/schemathesis/_hypothesis.py:81: CheckFailed
```

In the next alpha release, the failure would look like this:

```
est_csv_response_validation_direct.py F

============================================================= FAILURES ==============================================================
_________________________________________________ test_csv_validation[GET /success] _________________________________________________
+ Exception Group Traceback (most recent call last):
  |   File "/tmp/.../test_csv_response_validation_direct.py", line 46, in test_csv_validation
  |     case.validate_response(response, checks=(response_schema_conformance,))
  |   File "/.../src/schemathesis/generation/case.py", line 171, in validate_response
  |     raise FailureGroup(_failures, message) from None
  | schemathesis.core.failures.FailureGroup: Schemathesis found 2 distinct failures
  |
  | - Missing Content-Type header
  |
  |     The following media types are documented in the schema:
  |     - `application/json`
  |
  | - JSON deserialization error
  |
  |     Response must be valid JSON with 'Content-Type: application/json' header:
  |
  |       Expecting value: line 1 column 1 (char 0)
  |
  | [200] OK:
  |
  |     `name,age
  |     John,25`
  |
  | Reproduce with:
  |
  |     curl -X GET --insecure 'http://127.0.0.1:47371/api/success?format=csv'
  |
  |  (2 sub-exceptions)
  +-+---------------- 1 ----------------
    | schemathesis.openapi.checks.MissingContentType: Missing Content-Type header
    |
    | The following media types are documented in the schema:
    | - `application/json`
    +---------------- 2 ----------------
    | schemathesis.core.failures.MalformedJson: JSON deserialization error
    |
    | Response must be valid JSON with 'Content-Type: application/json' header:
    |
    |   Expecting value: line 1 column 1 (char 0)
    +------------------------------------
====================================================== short test summary info ======================================================
FAILED test_csv_response_validation_direct.py::test_csv_validation[GET /success]
========================================================= 1 failed in 0.06s =========================================================
```

The cURL sample includes all the things + there are no unnecessary traceback frames.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -19,7 +19,8 @@
 - Run positive and negative tests by default.
 - Add timestamps to report filenames to avoid rewriting exiting report in subsequent runs.
 - Coverage phase no longer emits duplicate query parameter names by default. The `phases.coverage.generate-duplicate-query-parameters` config option allows for controlling this behavior.
-- Report cURL commands for some network-level errors (connection resets, server crashes). [#2709](https://github.com/schemathesis/schemathesis/issues/2709) 
+- Report cURL commands for some network-level errors (connection resets, server crashes). [#2709](https://github.com/schemathesis/schemathesis/issues/2709)
+- Cleanup pytest output. [#2714](https://github.com/schemathesis/schemathesis/issues/2714)
 
 ### :fire: Removed
 

--- a/src/schemathesis/core/errors.py
+++ b/src/schemathesis/core/errors.py
@@ -441,7 +441,7 @@ def split_traceback(traceback: str) -> list[str]:
 
 
 def format_exception(
-    error: Exception,
+    error: BaseException,
     *,
     with_traceback: bool = False,
     skip_frames: int = 0,

--- a/src/schemathesis/generation/case.py
+++ b/src/schemathesis/generation/case.py
@@ -167,6 +167,7 @@ class Case:
                 curl=curl,
                 config=self.operation.schema.config.output,
             )
+            message += "\n\n"
             raise FailureGroup(_failures, message) from None
 
     def call_and_validate(

--- a/src/schemathesis/generation/hypothesis/builder.py
+++ b/src/schemathesis/generation/hypothesis/builder.py
@@ -9,7 +9,7 @@ from time import perf_counter
 from typing import Any, Callable, Generator, Mapping
 
 import hypothesis
-from hypothesis import Phase
+from hypothesis import Phase, Verbosity
 from hypothesis import strategies as st
 from hypothesis._settings import all_settings
 from hypothesis.errors import Unsatisfiable
@@ -96,6 +96,9 @@ def create_test(
 
     if settings.deadline == default.deadline:
         settings = hypothesis.settings(settings, deadline=DEFAULT_DEADLINE)
+
+    if settings.verbosity == default.verbosity:
+        settings = hypothesis.settings(settings, verbosity=Verbosity.quiet)
 
     if config.settings is not None:
         # Merge the user-provided settings with the current ones

--- a/src/schemathesis/pytest/plugin.py
+++ b/src/schemathesis/pytest/plugin.py
@@ -21,7 +21,9 @@ from schemathesis.core.errors import (
     InvalidRegexPattern,
     InvalidSchema,
     SerializationNotPossible,
+    format_exception,
 )
+from schemathesis.core.failures import FailureGroup
 from schemathesis.core.marks import Mark
 from schemathesis.core.result import Ok, Result
 from schemathesis.generation.hypothesis.given import (
@@ -245,6 +247,26 @@ def pytest_pycollect_makeitem(collector: nodes.Collector, name: str, obj: Any) -
         outcome.force_result(SchemathesisCase.from_parent(collector, test_function=obj, name=name, schema=schema))
     except Exception:
         outcome.get_result()
+
+
+@pytest.hookimpl(tryfirst=True)  # type: ignore[misc]
+def pytest_exception_interact(node: Function, call: pytest.CallInfo, report: pytest.TestReport) -> None:
+    if call.excinfo and call.excinfo.type is FailureGroup:
+        tb_entries = list(call.excinfo.traceback)
+        total_frames = len(tb_entries)
+
+        # Keep internal Schemathesis frames + one extra one from the caller
+        keep_from_index = 0
+        for i in range(total_frames - 1, -1, -1):
+            entry = tb_entries[i]
+
+            if "validate_response" in str(entry):
+                keep_from_index = max(0, i - 1)
+                break
+
+        skip_frames = keep_from_index
+
+        report.longrepr = "".join(format_exception(call.excinfo.value, with_traceback=True, skip_frames=skip_frames))
 
 
 @hookimpl(wrapper=True)

--- a/src/schemathesis/specs/openapi/schemas.py
+++ b/src/schemathesis/specs/openapi/schemas.py
@@ -659,6 +659,7 @@ class BaseOpenAPISchema(BaseSchema):
         return jsonschema.Draft4Validator
 
     def validate_response(self, operation: APIOperation, response: Response) -> bool | None:
+        __tracebackhide__ = True
         responses = {str(key): value for key, value in operation.definition.raw.get("responses", {}).items()}
         status_code = str(response.status_code)
         if status_code in responses:

--- a/test/cli/test_checks.py
+++ b/test/cli/test_checks.py
@@ -111,7 +111,6 @@ def test_negative_data_rejection_displays_all_cases(app_runner, cli, snapshot_cl
     assert (
         cli.run(
             f"http://127.0.0.1:{port}/openapi.json",
-            "-call",
             "--mode=all",
             "--phases=coverage",
             "--continue-on-failure",
@@ -120,6 +119,62 @@ def test_negative_data_rejection_displays_all_cases(app_runner, cli, snapshot_cl
                     "negative_data_rejection": {"expected-statuses": ["400", "401", "403", "404", "422", "428", "5xx"]}
                 }
             },
+        )
+        == snapshot_cli
+    )
+
+
+def test_format_parameter_csv_response(app_runner, cli, snapshot_cli):
+    raw_schema = {
+        "openapi": "3.0.0",
+        "paths": {
+            "/data": {
+                "get": {
+                    "parameters": [
+                        {
+                            "name": "format",
+                            "in": "query",
+                            "schema": {
+                                "type": "string",
+                                "enum": ["json", "csv"],
+                            },
+                        }
+                    ],
+                    "responses": {
+                        "200": {
+                            "description": "Data response",
+                            "content": {
+                                "application/json": {"schema": {"type": "object"}},
+                            },
+                        }
+                    },
+                }
+            }
+        },
+    }
+    app = Flask(__name__)
+
+    @app.route("/openapi.json")
+    def schema():
+        return jsonify(raw_schema)
+
+    @app.route("/data", methods=["GET"])
+    def data_endpoint():
+        format_param = request.args.get("format", "json")
+
+        if format_param == "csv":
+            return "name,age\nJohn,25", 200, {"Content-Type": ""}
+        return jsonify({"name": "John", "age": 25})
+
+    port = app_runner.run_flask_app(app)
+
+    assert (
+        cli.run(
+            f"http://127.0.0.1:{port}/openapi.json",
+            "-c response_schema_conformance",
+            "--mode=positive",
+            "--phases=fuzzing",
+            "--continue-on-failure",
         )
         == snapshot_cli
     )

--- a/test/test_pytest.py
+++ b/test/test_pytest.py
@@ -882,3 +882,58 @@ def test(case):
     result = testdir.runpytest()
     result.assert_outcomes(errors=1)
     assert "`test` has already been decorated with `override`" in result.stdout.str()
+
+
+def test_csv_response_validation_direct(testdir, openapi3_base_url):
+    testdir.make_test(
+        f"""
+import requests
+from schemathesis.core.transport import Response
+from schemathesis.specs.openapi.checks import response_schema_conformance
+
+schema.config.update(base_url="{openapi3_base_url}")
+schema.config.generation.update(modes=[GenerationMode.POSITIVE])
+
+@schema.include(name="GET /success").parametrize()
+def test_csv_validation(case):
+    request = requests.Request('GET', "http://127.0.0.1/api/success").prepare()
+
+    response = Response(
+        status_code=200,
+        headers={{}},
+        content=b"name,age\\nJohn,25",
+        request=request,
+        elapsed=0.1,
+        verify=False
+    )
+    case.validate_response(response, checks=(response_schema_conformance,))
+""",
+        paths={
+            "/success": {
+                "get": {
+                    "parameters": [
+                        {
+                            "name": "format",
+                            "in": "query",
+                            "schema": {"type": "string", "enum": ["json", "csv"]},
+                            "required": True,
+                        }
+                    ],
+                    "responses": {
+                        "default": {
+                            "description": "Data response",
+                            "content": {"application/json": {"schema": {"type": "object"}}},
+                        }
+                    },
+                }
+            }
+        },
+        schema_name="simple_openapi.yaml",
+    )
+    result = testdir.runpytest("-s")
+    result.assert_outcomes(failed=1)
+    result.stdout.re_match_lines(
+        [
+            r".*/api/success\?format=csv.*",
+        ]
+    )

